### PR TITLE
Premium v2

### PR DESCRIPTION
**Pull request by azeier**
Work in progress.

This is part 1 of 2 towards implementing https://github.com/HearthSim/HSReplay.net/issues/893.

This PR contains several assets that are going to be used in part 2.

## Before this can land:
- Merge https://github.com/HearthSim/loadtesting/pull/4

## Todo:
- [x] Add referrals back
- [x] Text pass
- [x] Mark new strings as translatable
- [x] ~~Move big assets to S3~~ Converting to jpg was good enough
- [x] Fix smoke tests (haven't verified they are broken, but they should be) 
- [x] Add actual images for testmonials

## Potential refactoring
-  Extract generic `Carousel` (responsible for rotating, takes data and a data => representation converter) component for `TesteomnialCarousel` and `ArchetypeHighlight`. Not necessarily as part of this PR.

## State as of 6/11
image

Mobile: [click here](https://user-images.githubusercontent.com/7737910/42614370-77f2747a-855a-11e8-830a-0ceefbcb4061.png)

### CODE REVIEW

File: hsreplaynet/static/scripts/src/components/premium/Testemonial.tsx [hunk 1]
```diff
@@ -0,0 +1,23 @@
+import React from "react";
+
+export interface TestemonialData {
```
Comment (joolean) [inline hsreplaynet/static/scripts/src/components/premium/Testemonial.tsx]: "Testimonial"

File: hsreplaynet/static/scripts/src/components/premium/Testemonial.tsx [hunk 2]
```diff
@@ -0,0 +1,23 @@
...
+	image: string;
+	name: string;
+	subtitle: string;
+	text: string;
+}
+
+export default class Testemonial extends React.Component<TestemonialData> {
```
Comment (joolean) [inline hsreplaynet/static/scripts/src/components/premium/Testemonial.tsx]: "Testimonial"

File: hsreplaynet/static/scripts/src/components/premium/TestemonialCarousel.tsx [hunk 3]
```diff
@@ -0,0 +1,109 @@
...
+import Testemonial, { TestemonialData } from "./Testemonial";
+
+interface Props {}
+
+interface State {
+	index: number;
+	lastIndex: number | null;
+}
+
+export default class TestemonialCarousel extends React.Component<Props, State> {
```
Comment (joolean) [inline hsreplaynet/static/scripts/src/components/premium/TestemonialCarousel.tsx]: "Testimonial"

File: hsreplaynet/static/scripts/src/pages/PremiumDetail.tsx [hunk 4]
```diff
@@ -20,6 +25,315 @@ interface State {}
...
+		if (UserData.hasFeature("new-premium")) {
+			return this.renderNew();
+		} else {
+			return this.renderOld();
+		}
+	}
+
+	private renderNew(): React.ReactNode {
+		const { hasSubscriptionPastDue, t } = this.props;
+		const isPremium = UserData.isPremium() || true;
```
Comment (azeier) [inline hsreplaynet/static/scripts/src/pages/PremiumDetail.tsx]: 🤔

**Comment by azeier** [comment]
Closing this in favor of just landing everything under #900.

**Pull request closed without merging**